src/corona_hakab_model/generation/circles.py

```diff
@@ -55,10 +56,33 @@ def get_indexes_of_my_circle(self, my_index):
         return rest_of_circle
 
     def get_snapshot(self):
-        return SocialCircleSnapshot(self.connection_type.name,len(self.agents))
+        return SocialCircleSnapshot(self.connection_type.name,len(self.agents),self.circle_id)
 
 
 @dataclass
 class SocialCircleSnapshot:
     type: str
-    num_members: int
\ No newline at end of file
+    num_members: int
+    id: int
+
+class SocialCircleConstraint:
+    def __init__(self,min_members,max_members,connection_type):
+        self.min_members = min_members
+        self.max_members = max_members
+        self.connection_type = connection_type
+
+    def meets_constraint(self,agent):
+        assert agent is not None
```

Comment:
Why? Let the normal error jump if agent is None...

Reply:
Done.